test/e2e/scenarios/Snapshot/snapshot_create_update_ref.yaml

```diff
@@ -0,0 +1,73 @@
+id: "SNAPSHOT_CREATE_UPDATE_REF"
+description: "In this test we create Snapshot and validate ref"
+#marks:
+#  - slow
+#  - blocked
+steps:
+  - steps:
+      - id: "CLUSTER_INITIAL_CREATE"
+        description: "Create cluster"
+        create:
+          apiVersion: $CRD_GROUP/$CRD_VERSION
+          kind: Cluster
+          metadata:
+            name: cluster$RANDOM_SUFFIX
+          spec:
+            name: cluster$RANDOM_SUFFIX
+            nodeType: db.t4g.small
+            aclName: open-access
+            numShards: 1
+        wait:
+          status:
+            conditions:
+              ACK.ResourceSynced:
+                status: "True"
+                timeout: 3600
+  - id: "SNAPSHOT_INITIAL_CREATE"
+    description: "Create snapshot"
+    create:
+      apiVersion: $CRD_GROUP/$CRD_VERSION
+      kind: Snapshot
+      metadata:
+        name: snapshot$RANDOM_SUFFIX
+      spec:
+        clusterName: cluster$RANDOM_SUFFIX
+        name: snapshot$RANDOM_SUFFIX
+    wait:
+      status:
+        conditions:
+          ACK.ResourceSynced:
+            status: "True"
+            timeout: 1800
+  - id: "SNAPSHOT_VALIDATE_REF"
+    description: "Validate ref"
+    patch:
+      apiVersion: $CRD_GROUP/$CRD_VERSION
+      kind: Snapshot
+      metadata:
+        name: snapshot$RANDOM_SUFFIX
+      spec:
+        clusterName:
+        clusterRef:
+          from:
+            name: cluster$RANDOM_SUFFIX
+    wait:
+      status:
+        conditions:
+          ACK.ReferencesResolved:
+            status: "True"
+            timeout: 100
```

Comment:
This operation has no useful effect. The only part of a `Snapshot` that you can update is the tags - https://github.com/aws-controllers-k8s/memorydb-controller/blob/main/pkg/resource/snapshot/hooks.go#L282-L297

A stronger test for this might just be to use the `clusterRef` rather than `clusterName` initially

Reply:
Sure, let me modify.